tools/check-workflow-run/main.go

```diff
@@ -0,0 +1,56 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/google/go-github/v61/github"
+	flag "github.com/spf13/pflag"
+)
+
+var (
+	headSHA           = flag.String("head-SHA", "", "the SHA for the workflow run")
+	workflowFileName  = flag.String("workflow", "build.yaml", "the workflow of interest")
+	generateProwError = flag.Bool("prow-error", true, "generate output that will stand out in prow")
+)
+
+// lookup useful information for a github action workflows most recent run
+// relating to a SHA.
+func main() {
+	flag.Parse()
+
+	if *headSHA == "" {
+		fmt.Println("--head-SHA is required")
+		os.Exit(2)
+	}
+
+	client := github.NewClient(nil)
+	ctx := context.Background()
```

Comment:
Just to make sure this does not get stuck in case of a weird network issue:
```suggestion
	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute())
	defer cancel()
```